source/common/router/header_parser_utils.cc

```diff
@@ -25,11 +25,11 @@ std::string HeaderParser::translateMetadataFormat(const std::string& header_valu
   const re2::RE2& re = getMetadataTranslatorPattern();
   ASSERT(re.ok());
   std::string new_header_value = header_value;
-  std::string json_array, metadata_type;
+  re2::StringPiece json_array, metadata_type;
   while (re.PartialMatch(new_header_value, re, &metadata_type, &json_array)) {
     std::string new_format;
     TRY_ASSERT_MAIN_THREAD {
-      Json::ObjectSharedPtr parsed_params = Json::Factory::loadFromString(json_array);
+      Json::ObjectSharedPtr parsed_params = Json::Factory::loadFromString(json_array.as_string());
```

Comment:
I was going to comment that we should convert the re2::StringPiece to a string_view and pass that way, when I noticed that Json::Factory::loadFromString takes a const std::string& rather than a string_view, so you can't do that. Fine.

But then I noticed that the implementation is broken:
https://github.com/envoyproxy/envoy/blob/7107d3b3592d1aab36caccb95b4ceed145e08c71/source/common/json/json_internal.cc#L678

That needs to use json.c_str() to get a nul-terminated char*. WDYT about throwing that into this PR?

Strings are so hard!

Reply:
Just pushed the fix to use `json.c_str()`. 

> Strings are so hard!

Agree 100%!